src/cloud_sync.h

```diff
@@ -62,6 +64,7 @@ class CloudSyncDestination
 
     /// Asynchronously uploads the file. Returned future throws on error.
     virtual dispatch::future<void> Upload(CatalogPtr file) = 0;
+    virtual bool Auth(wxWindow* parent) = 0;
```

Comment:
You are making this API depend on GUI code for reasons that aren't entirely clear to me (see the other comment; but this place shows why it bothers me).

Reply:
- That's description of [Crashed on open cached Crowdin file if logged out](68d5e5e54fe019ee877b40a49f938edefa8acdc5) commit (where this change is introduced) should be "Crashed on **saving of** cached Crowdin file if logged out" instead, my bad sorry.  I've implemented it similarly with how it's done in [here in `CrowdinOpenFile()`](https://github.com/tetatetit/poedit/blob/278a08250a71def13b71847c9e962309d829e90e/src/crowdin_gui.cpp#L552) and [here in `CrowdinSyncFile()`]( https://github.com/tetatetit/poedit/blob/278a08250a71def13b71847c9e962309d829e90e/src/crowdin_gui.cpp#L575)
1. But I could not put `CrowdinLoginDialog` as well as any other Crowdin specific code directly to closest using it GUI code  [`RunSync()`](https://github.com/tetatetit/poedit/blob/278a08250a71def13b71847c9e962309d829e90e/src/cloud_sync.h#L141) which assumed to be abstract/generalized for any `CloudSyncDestination` so knowing nothing about `CrowdinSyncDestination`
2. The only alternative I seen to avoid GUI code in `CrowdinSyncDestination` is to make `Auth()` to just `return CrowdinClient::Get().IsSignedIn()` silently without displaying `CrowdinLoginDialog` if not authorized. But in this case Poedit would just silently save such file locally, without bringing user attention that the file can be saved (uploaded) to Crowdin as well as providing option to sign in to Crowdin immediately "in place".

But please feel free to "Unresolve" this conversation if you see above variant 2 or anything else as better alternative.

Reply:
Forgot to specify steps to reproduce the crash:
1. Open any file from Crowdin in Poedit
2. Sign out from Crowdin without exiting (closing) Poedit (or probable might crash even if close and then re-open same file from local cache but don't remember exactly)
3. `Save` file (by `Ctrl+S` or `Save` in toolbar, or `File` \ `Save` - the way does not matter)